src/test/util_string_tests.cpp

```diff
@@ -8,28 +8,44 @@
 #include <test/util/setup_common.h>
 
 using namespace util;
+using util::detail::CheckNumFormatSpecifiers;
 
 BOOST_AUTO_TEST_SUITE(util_string_tests)
 
+template <unsigned NumArgs>
+void TfmFormatZeroes(const char* fmt)
+{
+    std::apply([fmt](auto... args) {
+        (void)tfm::format(std::string{fmt}, args...);
+    }, std::tuple_cat(std::array<int, NumArgs>{}));
+}
```

Comment:
nit: we could avoid `[fmt]` via `[&]` here, make it `static` and give it a `std::string` param directly:
```suggestion
template <unsigned NumArgs>
static void TfmFormatZeroes(const std::string& fmt)
{
    std::apply([&](auto... args) { tfm::format(fmt, args...); }, std::array<int, NumArgs>{});
}
```

Reply:
Thanks, taken in broad strokes.

Temporarily dded
```diff
BOOST_AUTO_TEST_CASE(ConstevalFormatString_NumSpec)
{
+    TfmFormatZeroes<10>("blah")
```
together with
```diff
void TfmFormatZeroes(const std::string& fmt)
{
    std::apply([&](auto... args) {
+        fprintf(stderr, "ARG COUNT: %d", sizeof...(args));
+        assert(false);
        (void)tfm::format(fmt, args...);
    }, std::array<int, NumArgs>{});
}
```
To verify that `tuple_cat` was unnecessary.